File: app/src/test/java/com/example/recipeclipper/fake/FakePageRecipeExtractor.kt
package com.example.recipeclipper.fake

import com.example.recipeclipper.data.PageRecipeExtractor
import com.example.recipeclipper.data.model.PageSelection

/**
 * The on-device model reading a page (#103), faked: [picks] is what it "picks" from any text,
 * [languages] the languages it reads, with a window of [windowChars]. [asked] records every
 * text it was given.
 */
class FakePageRecipeExtractor(
    var picks: PageSelection? = null,
    var languages: Set<String> = setOf("en"),
    var windowChars: Int = 6_000
) : PageRecipeExtractor {
    val asked = mutableListOf<String>()

    override suspend fun windowChars(language: String): Int? = windowChars.takeIf { language in languages }

    override suspend fun extract(text: String, language: String): PageSelection? {
        asked += text
        return picks
    }
}

File: ios/RecipeClipperTests/Fakes/FakePageRecipeExtractor.swift
@testable import RecipeClipper

/// The on-device model reading a page (#103), faked: `picks` is what it "picks" from any text,
/// `languages` the languages it reads, with a window of `windowChars`. `asked` records every
/// text it was given. Android's `FakePageRecipeExtractor`.
final class FakePageRecipeExtractor: PageRecipeExtractor {
    var picks: PageSelection?
    var languages: Set<String>
    var chars: Int
    private(set) var asked: [String] = []

    init(picks: PageSelection? = nil, languages: Set<String> = ["en"], windowChars: Int = 6_000) {
        self.picks = picks
        self.languages = languages
        self.chars = windowChars
    }

    func windowChars(language: String) async -> Int? { languages.contains(language) ? chars : nil }

    func extract(_ text: String, language: String) async -> PageSelection? {
        asked.append(text)
        return picks
    }
}
